sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIService.scala

```diff
@@ -52,8 +52,22 @@ private[hive] class SparkSQLCLIService(hiveServer: HiveServer2, sqlContext: SQLC
 
     if (UserGroupInformation.isSecurityEnabled) {
       try {
-        HiveAuthFactory.loginFromKeytab(hiveConf)
-        sparkServiceUGI = Utils.getUGI()
+        val principal = hiveConf.getVar(ConfVars.HIVE_SERVER2_KERBEROS_PRINCIPAL)
+        val keyTabFile = hiveConf.getVar(ConfVars.HIVE_SERVER2_KERBEROS_KEYTAB)
+        if (principal.isEmpty || keyTabFile.isEmpty) {
+          throw new IOException(
+            "HiveServer2 Kerberos principal or keytab is not correctly configured")
+        }
+
+        val originalUgi = UserGroupInformation.getCurrentUser
```

Comment:
just curious, why here we are using `UserGroupInformation.getCurrentUser` and a few lines after `Utils.getUGI()`? I see that they basically do the same, but I am  interested if there are reasons for doing this.

Reply:
I don't think there's any particular reason, we just copy what HS2 did before.

Reply:
Assuming I understood your question @mgaido91, we use Utils.getUGI only when we do an explicit login; else fallback on originalUgi (the current user).

This is because after a login, the instance of ugi changes from underneath us in UserGroupInformation class.

Reply:
@mridulm thanks for your answer. Actually my question was a bit different, but @jerryshao answered me: which is the difference between `UserGroupInformation.getCurrentUser` and `Utils.getUGI()`? Since `Utils.getUGI()` does the same unless `HADOOP_USER_NAME` is set, my feeling was that using either one or the other function is the same here, so I just wondered why using one on one place and the other few lines after and not always the same one: I just wanted to make sure that they are actually the same. Thanks for you kind answers.

Reply:
Ah ok, Utils.getUGI handles proxying in HS2 iirc - this is, currently, not relevant in our case imo.
You are right, we could have simply used `Utils.getUGI` instead of `UserGroupInformation.getCurrentUser` (it would have been more appropriate as well).
Something to keep in mind in case users are relying on HADOOP_USER_NAME for STS (highly doubt it, but you never know !)

Reply:
So shall we change to `Utils.getUGI`? Basically it is the same. But as @mridulm mentioned, someone may set this "HADOOP_USER_NAME", though doAs is not worked in STS.